crates/cli/src/utils/cmd.rs

```diff
@@ -416,3 +417,36 @@ pub async fn print_traces(result: &mut TraceResult, decoder: &CallTraceDecoder)
     println!("Gas used: {}", result.gas_used);
     Ok(())
 }
+
+/// Traverse the artifacts in the project to generate local signatures and merge them into the cache file.
+pub fn cache_local_signatures(output: &ProjectCompileOutput, cache_path: PathBuf) -> Result<()> {
+    let mut cached_signatures = CachedSignatures::default();
+    let path = cache_path.join("signatures");
+    if !path.is_file() {
+        std::fs::create_dir_all(&cache_path)?;
+    } else {
+        let cache_contents = std::fs::read_to_string(path.clone()).unwrap();
+        match serde_json::from_str::<CachedSignatures>(&cache_contents) {
+            Ok(existed_signatures) => cached_signatures = existed_signatures,
+            Err(e) => {
+                println!("parse cached local signatures file error: {}", e);
+            }
+        }
+        dbg!(&cached_signatures);
```

Comment:
this is now duplicated with `SignatureIdentifier::new`, can we move this logic to `CachedSignatures::load(path)`